blog/_posts/2019-03-02-custom-primitiv-type.md

```diff
@@ -0,0 +1,383 @@
+---
+layout: post
+title: Declare primitive type in Julia
+author: 鄭景文（Peter Cheng）, 杜岳華（Yueh-Hua Tu）
+---
+
+Julia provide some APIs for you to declare your own primitive types in Julia. However, when you open [the official docs about primitive type](https://docs.julialang.org/en/v1/manual/types/#Primitive-Types-1), you still don't know how to make your new primitive types work in your code. Therefore, here is a simple tutorial that tell you how to do so.
+
+
+## Introduction
+
+If you want define a primitve type, first, you have to declare it with the name and the number of bits it take with the `primitive type` keyword. 
+
+    primitive type «name» «bits» end
+    primitive type «name» <: «supertype» «bits» end
+
+Currently, Julia only support sizes that are multiples of 8 bits. So like the `Bool` only take 1 bit, you still need to declare it as 8 bits. For example:
+
+```julia
+    primitive type Bool <: Integer 8 end
+```
+
+Up to here, we have almost everything from the docs. Next, we are going to show how 
+to use our own primitive type with a simple example.
+
+
+## The problem
+
+After declaring our new primitive type, that's say we do the following:
+
+```julia
+    primitive type A 72 end
+```
+
+But soon you will find that you don't know how to create a instance of `A` because Julia don't have a default constructor for a new primitive type. And if you think further, you will notice that there seems to be no methods for you to manipulate those instances. You can't even do bitwise addition becuase `+` donsn't work on our new primitive type.
```

Comment:
```suggestion
But soon you will find that you don't know how to create a instance of `A` because Julia does not have a default constructor for primitive types. And if you think further, you will notice that there seems to be no methods for you to manipulate those instances. You can't even do bitwise addition becuase `+` does not work on our new primitive type.
```